### ISSUE TITLE
Migrate to the new PostgreSQL

### ISSUE THREAD

**Issue by skabashnyuk** [eclipse-che/che#20246]
### Is your task related to a problem? Please describe.
The final Release of PostgreSQL 9.6 should be November 11, 2021.
We use it as a database for che-server and keycloak.
We need to migrate to a more recent common version.

### Describe the solution you'd like
- Figure out the latest PostgreSQL we need.
- Pick docker image
- Push docker image to http://quay.io/
- Provide a strategy for new installations
- Provide a strategy for old installations

### Describe alternatives you've considered
n/a

### Imeplentation
 - [x] Figure out the latest PostgreSQL we need. Sync with @AndrienkoAleksandr  about https://github.com/eclipse/che/issues/19585 and check if PostgreSQL 13 is acceptable. Test images with this property of CheCluster CR https://github.com/eclipse-che/che-operator/blob/a3abf33b108adbf67ea8abeccb17849f573fac16/api/v1/checluster_types.go#L452
 - [x] Pick docker image. Proposal "postgres:13.3-alpine"
 - [x] Check if TCK of che-server is passing with "postgres:13.3-alpine"
 - [x] Sync with @benoitf and @nickboldt to get an agreement with the selected image.
 - [x] push image to quay.io/eclipse/che--centos--postgresql-13-centos7:1-71b24684d64da46f960682cc4216222a7e4ed8b1a31dd5a865b3e71afdea20d2
 - [x] Adapt the code of che-operator. New installation should use PostgreSQL 13. Old installation should remain untouched. https://github.com/eclipse/che/issues/20398
 - [x] Provide a migration guidance from PostgreSQL 9 -> 13. Proposal is to use Che's build in backup/restore methods. https://github.com/eclipse/che/issues/20398
 - [x] Change the image PostgreSQL in integration tests of che server https://github.com/eclipse-che/che-server/pull/74

 
### Additional context
  - https://www.postgresql.org/support/versioning/
  - https://github.com/eclipse/che/issues/18292
  - https://github.com/eclipse-che/che-server/tree/main/dockerfiles/postgres
 - https://github.com/eclipse/che/pull/16903

**Comment by skabashnyuk** [comment]
Looks like  `docker.io/postgres:13.3-alpine3.14` upstream and `registry.redhat.io/rhel8/postgresql-13` for downstream is our choice. Have an agreement from @benoitf and @nickboldt

**Comment by skabashnyuk** [comment]
>https://github.com/eclipse-che/che-server/tree/main/dockerfiles/postgres
>I think it is used in helmcharts

We definitely need to use the same image across all possible installation ways.

**Comment by xbaran4** [comment]
Is there a possibility we could use centos/postgresql-13-centos7 in https://github.com/eclipse-che/che-server/blob/72bfc85436022bc87aad109829411ef7f9b48663/dockerfiles/postgres/Dockerfile#L9 ?

Since it is pretty much one line change and all the configuration stays the same, so everything works out of the box.
CC @nickboldt  @skabashnyuk @benoitf

**Comment by skabashnyuk** [comment]
>Is there a possibility we could use centos/postgresql-13-centos7 in https://github.com/eclipse-che/che-server/blob/72bfc85436022bc87aad109829411ef7f9b48663/dockerfiles/postgres/Dockerfile#L9 ?

For testing - yes. I see only "linux/amd64" arch which might be problematic.

**Comment by xbaran4** [comment]
[Postgres 13 alpine and buster images](https://hub.docker.com/_/postgres) have different configuration from the centos and rhel prostgres 9.6 we use now (different env variables, different directory layout, different entrypoint scripts and possibly more).

If we decided for alpine or buster in Che, we would need to adapt che operator for the new image and that would mean having different implementations in che operator for upstream and downstream.

So we decided to go with [centos/postgresql-13-centos7](https://hub.docker.com/r/centos/postgresql-13-centos7) for upstream and [rhel8/postgresql-13](https://catalog.redhat.com/software/containers/rhel8/postgresql-13/5ffdbdef73a65398111b8362?container-tabs=overview&gti-tabs=registry-tokens) for downstream, since they are sharing configuration.
